test/browser.ts

```diff
@@ -48,6 +48,49 @@ describe('libp2p-webtransport', () => {
 
     const err = await expect(node.dial(ma)).to.eventually.be.rejected()
     expect(err.errors[0].toString()).to.contain('WebTransportError: Opening handshake failed.')
+
+    await node.stop()
+  })
+
+  it('Closes writes of streams after they have sunk a source', async () => {
+    // This is the behavor of stream muxers: (see mplex, yamux and compliance tests: https://github.com/libp2p/js-libp2p-interfaces/blob/master/packages/interface-stream-muxer-compliance-tests/src/close-test.ts)
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const maStr: string = process.env.serverAddr!
+    const ma = multiaddr(maStr)
+    const node = await createLibp2p({
+      transports: [webTransport()],
+      connectionEncryption: [() => new Noise()]
+    })
+
+    async function * gen () {
+      yield new Uint8Array([0])
+      yield new Uint8Array([1, 2, 3, 4])
+      yield new Uint8Array([5, 6, 7])
+      yield new Uint8Array([8, 9, 10, 11])
+      yield new Uint8Array([12, 13, 14, 15])
+    }
+
+    await node.start()
+    const stream = await node.dialProtocol(ma, 'echo')
+
+    await stream.sink(gen())
+
+    let expectedNextNumber = 0
+    for await (const chunk of stream.source) {
+      for (const buf of chunk) {
+        for (const byte of buf) {
+          expect(byte).to.eql(expectedNextNumber++)
+        }
+      }
+    }
+    expect(expectedNextNumber).to.eql(16)
```

Comment:
It's a tiny nit, but `.eql` is shorthand for `deep.equal` which isn't necessary for numbers.

```suggestion
    expect(expectedNextNumber).to.equal(16)
```

Reply:
I new about the difference, but didn't know about `deep.equal`. Thanks